rolecall/rolecall.py

```diff
@@ -101,73 +46,166 @@ class RoleCall:
 
-    @roleboard.command(pass_context=True, name="channel", no_pm=True)
-    async def roleboard_channel(self, ctx, channel: discord.Channel=None):
-        """Set the roleboard for this server.
+    @rolecall.command(pass_context=True, name="add", no_pm=True)
+    async def rolecall_add(self, ctx, channel: discord.Channel, 
+                            content_or_message_id: str, role: str, 
+                            emoji: str, 
+                            private_channel: str = None,
+                            ):
+        """
+        Add a role to a message. 
+        
+        channel
+
+           The channel where the new message will be posted or the channel
+           where the existing message is located
 
-        Leave blank to turn off the roleboard
+        content_or_message_id
+
+           Contents of the message. If the message
+           already exists, provide the message id instead.
+
+        role 
+
+           Name of the role. If a non-existing role is provided, 
+           it will be created for you.
+
+        emoji 
+
+           Emoji corresponding to the role which users will click
+           on.
+ 
+        private_channel(Optional) 
+
+           A private channel that members of the role will be granted access to. 
+           If a non-existing channel is provided, it will be created for you.
         """
         server = ctx.message.server
-        #channel = ctx.message.channel
         author = ctx.message.author
 
-        """
-        TODO: limit amount of messages. when over,
-            log warning and DM server owner /
-            infractor (if he's adding a new invite)
-        ^ invalid todo due to updated implementation notes
-        """
+        # retrieve role mentions in the command message
+        roles = ctx.message.raw_role_mentions
+        if len(roles) == 1:
+            role_object = discord.utils.get(server.roles, id=roles[0])
+        else:
+            role_object = await self.get_or_create("role", role, server)
 
-        settings = self.settings[server.id]
+        # retrieve channel mentions in the command message
+        channels = ctx.message.raw_channel_mentions
+        if len(channels) == 2:
+            role_channel = self.bot.get_channel(channels[1])
+        else:
+            role_channel = private_channel
 
-        if channel is None and not \
-           await self.prompt(ctx, "turn off the roleboard? (yes/no)"):
-            if settings["ROLEBOARD"] is None:
-                rb_channel = None
-            else:
-                rb_channel = get_channel_by_name(server, settings["ROLEBOARD"])
-            await self.bot.say('Ok. Roleboard is still {}'
-                               .format(rb_channel and rb_channel.mention))
+        # create the role's personal channel
+        try:
+            await self.create_or_edit_role_channel(server, role_object, role_channel)
+        except Exception as e:
+            print(e)
```

Comment:
be careful with catch-all try/excepts.
also, to be nice to your users, you might want to use Red's logger instead of printing the exception directly to the console